### PR TITLE
[WIP] Transformer tutorial

### PR DESCRIPTION
**Pull request by zhangguanheng66**
Add a tutorial for nn.Transformer module in PyTorch.
Train a transformer encoder for word language modeling.

### CODE REVIEW

**Comment by zhangguanheng66** [comment]
@brianjo @SethHWeidman @jlin27

File: beginner_source/transformer_tutorial.py [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+import math
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+class TransformerModel(nn.Module):
+
+    def __init__(self, ntoken, ninp, nhead, nhid, nlayers, dropout=0.5):
+        super(TransformerModel, self).__init__()
+        from torch.nn import TransformerEncoder, TransformerEncoderLayer
```
Comment (SethHWeidman) [inline beginner_source/transformer_tutorial.py]: Is there a reason this import is inside the `Module`?

**Comment by SethHWeidman** [comment]
Looks good! I'll make some minor grammatical fixes, but these shouldn't block the tutorial from landing. Thanks George!

File: beginner_source/transformer_tutorial.py (hunk 1 above)
Comment (zhangguanheng66) [reply beginner_source/transformer_tutorial.py]: not really. I could move them out of the class.

**Pull request merged**